imap_processing/codice/constants.py

```diff
@@ -719,6 +764,29 @@
     },
 }
 
+# Define the packet fields needed to be stored in segmented data and their
+# corresponding bit lengths for direct event data products
+IAL_BIT_STRUCTURE = {
+    "SHCOARSE": 32,
+    "PACKET_VERSION": 16,
+    "SPIN_PERIOD": 16,
+    "ACQ_START_SECONDS": 32,
+    "ACQ_START_SUBSECONDS": 20,
```

Comment:
I thought you had only a acquisition_time field per your above comments/code?

Should these be in Laura's root packet definition instead so you can unpack them as `CODICE_PLAN_ID` etc.?

Do you have SHCOARSE transmitted in the CoDICE portion of the packet? I thought the entire packet just had one SHCOARSE associated with it for all instrument to reference.

Reply:
Thanks for pointing this out. It got me thinking more about this and indeed I should have been calculating `epoch` based on `acq_start_seconds` and `acq_start_subseconds` instead of `shcoarse`.

Each CoDICE I-ALiRT packet has all three of `SHCOARSE`, `ACQ_START_SECONDS`, and `ACQ_START_SUBSECONDS`, but the CoDICE team prefers to se the acq starts to determine epoch instead of `SHCOARSE`.

Since these are all stored in the `DATA_xx` fields of the packets themselves, they need to be parsed directly from the bit strings of the `DATA_xx` fields, and thus I am not sure that could be easily captured in a XML packet definition(?)

Reply:
This is slightly unclear to me as well. I-ALiRT is a bit odd because the packet definition is per a single event, but you are stacking DATA_xx fields across multiple packets. So the definitions are indeed a bit odd and likely not helpful. So I agree with your approach here for now 👍